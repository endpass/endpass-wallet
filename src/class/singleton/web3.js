// Wrapper around global web3 instance
import Web3 from 'web3';
import { ProviderFactory, Network } from '@endpass/class';

const defaultProvider = ProviderFactory.create(
  Network.DEFAULT_NETWORKS[Network.NET_ID.MAIN].url,
);

const createWeb3Instance = provider => {
  const web3 = new Web3();

  web3.setProvider(provider);

  return web3;
};

const web3 = createWeb3Instance(defaultProvider);

// Make web3 global for integration tests
if (window.Cypress) {
  window.web3 = web3;
}

export default web3;
