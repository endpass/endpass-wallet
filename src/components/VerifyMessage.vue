<template>
  <v-form
    data-test="verifying-message-form"
    @submit="verifyMessage"
  >
    <v-textarea
      v-model="signedMessageString"
      label="Signature"
      data-test="signed-message-textarea"
    />
    <v-button
      :disabled="!signedMessage"
      class-name="is-primary is-medium"
      data-test="verify-button"
    >Verify message</v-button>
    <div
      v-if="address"
      class="field"
      data-test="address-field"
    >
      <label class="label">Address</label>
      <p>{{ address }}</p>
    </div>
  </v-form>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'VerifyMessage',

  data: () => ({
    address: null,
    signedMessageString: null,
  }),

  computed: {
    ...mapGetters('accounts', ['wallet']),

    signedMessage() {
      try {
        return JSON.parse(this.signedMessageString);
      } catch (error) {
        return '';
      }
    },
  },

  methods: {
    async verifyMessage() {
      try {
        const { message, signature } = this.signedMessage;
        this.address = await this.wallet.recover(message, signature);
      } catch (error) {
        this.address = null;

        this.$notify({
          title: 'Error while verifying the message',
          text:
            'An error occurred while verifying the message. Please try again.',
          type: 'is-danger',
        });
        /* eslint-disable-next-line no-console */
        console.error(error);
      }
    },
  },
};
</script>
