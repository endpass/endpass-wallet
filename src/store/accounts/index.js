import actions from './actions';
import getters from './getters';
import mutations from './mutations';

/**
 * @namespace
 * @property {Boolean} isInited Status of initializing accounts module
 * @property {Object} hdKey The encrypted xprv key for the user's hd wallet as a V3 keystore object.
 *  All accounts are direct child accounts.
 * @property {Object} hadrwareXpub The user's hardware xpub, keyed by wallet type
 * @property {Object} wallets The user's wallets, mapped by address
 * @property {String} address Current wallet address
 * @property {String} balance Currect account balance
 */
const state = {
  isInited: false,
  hardwareXpub: {},
  wallets: {},
  hdKey: null,
  address: null,
  balance: null,
};

export default {
  namespaced: true,
  state,
  actions,
  getters,
  mutations,
};
