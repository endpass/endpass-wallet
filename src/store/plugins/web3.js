import Web3 from 'web3';

import { ProviderFactory, web3 } from '@/class';

const setProvider = ({ dispatch }, provider) => {
  web3.setProvider(provider);

  if (provider.startPollingNewBlockHeaders) {
    provider.startPollingNewBlockHeaders(
      web3.eth.getBlockNumber,
      web3.eth.getBlock,
    );
  }

  dispatch('web3/subscribeOnBlockUpdates');
};

const getProvider = (store, url) => {
  const newProvider = ProviderFactory.create(url);

  if (newProvider.getFallbackProviders && newProvider.setErrorHandler) {
    const errorHandler = async () => {
      const fallbackProviders = newProvider.getFallbackProviders();
      let fallbackProvider = null;

      for (let i = 0; i < fallbackProviders.length; i += 1) {
        try {
          const web3Temp = new Web3(fallbackProviders[i]);

          // This can be very long if the server is not responding for a long time(about 2 min)
          /* eslint-disable-next-line no-await-in-loop */
          await web3Temp.eth.net.getId();
          fallbackProvider = fallbackProviders[i];
          break;
          /* eslint-disable-next-line no-empty */
        } catch (error) {}
      }

      if (fallbackProvider) {
        setProvider(store, fallbackProvider);
      }
    };

    newProvider.setErrorHandler(errorHandler);
  }

  return newProvider;
};

export default store => {
  store.subscribe(({ type, payload }) => {
    if (type === 'web3/CHANGE_NETWORK' && !window.Cypress) {
      setProvider(store, getProvider(store, payload.url));
    }
  });
};
