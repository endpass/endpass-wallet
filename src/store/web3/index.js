import { Network } from '@endpass/class';
import actions from './actions';
import getters from './getters';
import mutations from './mutations';

const state = {
  storedNetworks: [],
  blockNumber: null,
  handledBlockNumber: null,
  activeNet: {},
  interval: null,
  activeCurrency: Network.CURRENCIES[0],
};

export default {
  namespaced: true,
  state,
  actions,
  getters,
  mutations,
};
