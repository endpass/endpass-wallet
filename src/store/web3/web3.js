import Web3 from 'web3';
import { infuraConf } from '@/config.js';
import EthBlockTracker from 'eth-block-tracker';
import storage from '@/services/storage';
import { subscribtionsBlockchainInterval } from '@/config'
import { providerFactory } from '@/class';

const activeNet = {
  name: 'Main',
  id: 1,
  networkType: 'main',
  url: `https://mainnet.infura.io/${infuraConf.key}`,
};

const provider = providerFactory(activeNet.url);
const web3 = new Web3(provider);

export default {
  namespaced: true,
  state() {
    return {
      web3,
      defaultNetworks: [
        {
          id: 1,
          networkType: 'main',
          name: 'Main',
          url: `https://mainnet.infura.io/${infuraConf.key}`,
        },
        {
          name: 'Ropsten',
          networkType: 'ropsten',
          id: 3,
          url: `https://ropsten.infura.io/${infuraConf.key}`,
        },
        {
          name: 'Rinkeby',
          networkType: 'rinkeby',
          id: 4,
          url: `https://rinkeby.infura.io/${infuraConf.key}`,
        },
      ],
      storedNetworks: [],
      isSyncing: false,
      blockNumber: 0,
      activeNet,
    };
  },
  getters: {
    networks(state) {
      return state.defaultNetworks.concat(state.storedNetworks);
    },
  },
  mutations: {
    changeNetwork(state, network) {
      state.activeNet = network;
      const provider = providerFactory(state.activeNet.url)
      if (state.web3 && state.web3.setProvider) {
        if (state.web3.currentProvider.destroy) {
          state.web3.currentProvider.destroy()
        }

        state.web3.setProvider(provider);
      } else {
        state.web3 = new Web3(provider);
      }
    },
    addNewProvider(state, network) {
      state.storedNetworks.push(network);
      storage.write('networks', state.storedNetworks);
    },
    setProviders(state, networks) {
      state.storedNetworks = networks;
    },
    setSyncStatus(state, syncObject) {
      state.isSyncing = syncObject;
    },
    setBlockNumber(state, number) {
      state.blockNumber = number;
    },
    setNetworkType(state, type) {
      state.activeNet.networkType = type;
    },
  },
  actions: {
    changeNetwork({ commit, dispatch, getters }, networkId) {
      const netIndex = getters.networks.findIndex(net => net.id === networkId);
      const network = getters.networks[netIndex];
      commit('changeNetwork', network);
      storage.write('net', network.id);
      return Promise.all([
        dispatch('fetchNetworkType'),
        dispatch('subscribeOnBlockUpdates'),
        dispatch('tokens/subscribeOnTokenUpdates',{}, {root: true})
      ]);
    },
    addNewProvider({ commit, dispatch, getters }, network) {
      network.id = getters.networks.reduce((max, next) => {
        return max.id > next.id ? max : next;
      }).id + 1;
      commit('addNewProvider', network);
      return dispatch('changeNetwork', network.id);
    },
    subscribeOnSyncStatus(context) {
      let providerCache = context.state.web3.currentProvider;
      let promise = context.state.web3.eth.isSyncing();
      promise.then(resp => {
        if (providerCache === context.state.web3.currentProvider) {
          context.commit('setSyncStatus', resp);
          setTimeout(()=> {
            context.dispatch('subscribeOnSyncStatus')
          }, subscribtionsBlockchainInterval);
        } else {
          context.dispatch('subscribeOnSyncStatus')
        }
      });
      return promise;
    },
    fetchNetworkType(context) {
      // network type already set, return resolved promise
      if (context.state.activeNet.networkType) {
        return Promise.resolve();
      }
      let promise = context.state.web3.eth.net.getNetworkType();
      promise.then(resp => {
        context.commit('setNetworkType', resp);
      });
      return promise;
    },
    subscribeOnBlockUpdates(context) {
      if (context.state.blockSubscribtion) {
        context.state.blockSubscribtion.stop();
      }
      context.state.blockSubscribtion = new EthBlockTracker({
        provider: context.state.web3.currentProvider,
        pollingInterval:subscribtionsBlockchainInterval
      });
      context.state.blockSubscribtion.on('latest', block => {
        if (context.rootState.accounts.activeAccount) {
          context.dispatch('accounts/updateBalance', {}, { root: true });
        }
        context.commit(
          'setBlockNumber',
          web3.utils.hexToNumberString(block.number)
        );
      });
      context.state.blockSubscribtion.start();
    },
    init({ commit, dispatch, state }) {
      return storage
        .read('net')
        .then(cachedNet => parseInt(cachedNet))
        .then(cachedNet =>
          Promise.all([
            storage.read('networks').then(net => (net ? net : [])),
            cachedNet ? cachedNet : 1,
          ])
        )
        .then(([storedNetworks, cachedNet]) => {
          const activeNet =
            state.defaultNetworks.find(net => net.id === cachedNet) ||
            storedNetworks.find(net => net.id === cachedNet);

          commit('setProviders', storedNetworks || []);
          commit('changeNetwork', activeNet);
          dispatch('subscribeOnSyncStatus');
          dispatch('subscribeOnBlockUpdates');
        })
        .catch(e => console.error(e));
    },
  },
};
