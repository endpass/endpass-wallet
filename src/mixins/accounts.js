// Mixin for common methods for Ethereum accounts
import web3 from 'web3';

const { fromWei } = web3.utils;

export default {
  computed: {
    address() {
      return this.$store.state.accounts.address;
    },

    balance() {
      return this.$store.state.accounts.balance === null
        ? null
        : fromWei(this.$store.getters['accounts/balance']);
    },
  },
};
