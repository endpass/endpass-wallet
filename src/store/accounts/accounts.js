import { userService } from '@/services';
import storage from '@/services/storage';
import web3 from 'web3';
import Bip39 from 'bip39';
import HDKey from 'ethereumjs-wallet/hdkey';
import { hdKeyMnemonic, kdfParams } from '@/config';
import EthWallet from 'ethereumjs-wallet';
import { SAVE_TOKENS } from '@/store/tokens/mutations-types';
import { Wallet, Address } from '@/class';
import { BigNumber } from 'bignumber.js';
import keystore from '@/utils/keystore';

export default {
  namespaced: true,
  state: {
    // TODO move to user store?
    email: null,
    // The encrypted xprv key for the user's hd wallet as a V3 keystore
    // object. All accounts are direct child accounts.
    hdKey: null,
    // The user's wallets, keyed by address
    wallets: {},
    // The currently selected wallet
    wallet: null,
    address: null,
    balance: null,
    price: null,
    // prettier-ignore
    availableCurrencies: ['USD', 'AUD', 'BRL', 'CAD', 'CHF', 'CLP', 'CNY', 'CZK', 'DKK', 'EUR', 'GBP', 'HKD', 'HUF', 'IDR', 'ILS', 'INR', 'JPY', 'KRW', 'MXN', 'MYR', 'NOK', 'NZD', 'PHP', 'PKR', 'PLN', 'RUB', 'SEK', 'SGD', 'THB', 'TRY', 'TWD', 'ZAR'],
    settings: {
      fiatCurrency: 'USD',
    },
    otpSettings: {
      secret: null,
      status: null,
    },
  },
  getters: {
    isPublicAccount(state) {
      return (
        state.address instanceof Address && !(state.wallet instanceof Wallet)
      );
    },
    // TODO move to user store?
    isLoggedIn(state) {
      return !!state.email;
    },
    balance(state, getters, rootState, rootGetters) {
      if (state.balance === null) return null;

      const pendingBalance = rootGetters['transactions/pendingBalance'];
      const balanceWei = BigNumber(state.balance || '0')
        .minus(pendingBalance)
        .toFixed();

      return web3.utils.fromWei(balanceWei);
    },
    // Returns a decrypted HD Wallet
    hdWallet: state => password => {
      if (!state.hdKey) {
        return null;
      }
      return keystore.decryptHDWallet(password, state.hdKey);
    },
  },
  mutations: {
    setAddress(state, addressString) {
      state.address = new Address(addressString);
    },
    setWallet(state, wallet) {
      state.wallet = wallet;
    },
    addWallet(state, walletV3) {
      const wallet = new Wallet(walletV3);
      const { address } = walletV3;
      state.wallets = {
        ...state.wallets,
        [address]: wallet,
      };
    },
    // Saves the encrypted HD wallet key in V3 keystore format
    // Formerly addHdWallet
    setHdKey(state, key) {
      state.hdKey = key;
    },
    setBalance(state, balance) {
      state.balance = balance;
    },
    setPrice(state, price) {
      state.price = price;
    },
    setSettings(state, settings) {
      state.settings = JSON.parse(JSON.stringify(settings));
    },
    setEmail(state, email) {
      state.email = email;
    },
    setOtpSettings(state, otpSettings) {
      state.otpSettings = otpSettings;
    },
  },
  actions: {
    selectWallet({ commit, state, dispatch }, address) {
      commit('setWallet', state.wallets[address]);
      commit('setAddress', address);
      return dispatch(
        'tokens/subscribeOnTokensBalancesUpdates',
        {},
        { root: true },
      );
    },
    addWallet({ commit, dispatch, state }, json) {
      json.address = web3.utils.toChecksumAddress(json.address);
      commit('addWallet', json);

      return userService
        .setAccount(json.address, json)
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    addWalletAndSelect({ dispatch }, json) {
      return dispatch('addWallet', json)
        .then(() =>
          dispatch('selectWallet', web3.utils.toChecksumAddress(json.address)),
        )
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    // Import wallet from json V3 keystore
    async addWalletWithV3({ commit, dispatch }, { json, password }) {
      try {
        const wallet = new Wallet(json);
        let privateKey = await wallet.getPrivateKey(password);
        return dispatch('addWalletWithPrivateKey', privateKey);
      } catch (e) {
        return dispatch('errors/emitError', e, { root: true });
      }
    },
    async addWalletWithPrivateKey(
      { commit, dispatch },
      { privateKey, password },
    ) {
      try {
        const wallet = EthWallet.fromPrivateKey(Buffer.from(privateKey, 'hex'));
        const json = keystore.encryptWallet(password, wallet);
        return dispatch('addWalletAndSelect', json);
      } catch (e) {
        return dispatch('errors/emitError', e, { root: true });
      }
    },
    generateWallet({ commit, dispatch, state, getters }, password) {
      if (!state.hdKey) {
        return;
      }
      try {
        let hdWallet = getters.hdWallet(password);
        let i = Object.keys(state.wallets).length;
        let wallet = hdWallet.deriveChild(i).getWallet();
        let json = keystore.encryptWallet(password, wallet);
        return dispatch('addWalletAndSelect', json);
      } catch (e) {
        return dispatch('errors/emitError', e, { root: true });
      }
    },
    // Saves HD wallet's extended keys on the server
    saveHdWallet({ commit, dispatch, state }, json) {
      return userService.setAccount(json.address, json);
    },
    addHdWallet({ commit, dispatch }, { key, password }) {
      const seed = Bip39.mnemonicToSeed(key);
      const hdKey = HDKey.fromMasterSeed(seed);
      const hdWallet = hdKey.derivePath(hdKeyMnemonic.path);
      // Encrypt extended private key
      const json = keystore.encryptHDWallet(password, hdWallet);
      commit('setHdKey', json);

      // Save HD keys and generate the first child wallet
      return dispatch('saveHdWallet', json)
        .then(() => dispatch('generateWallet', password))
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    async addMultiHdWallet({ commit, dispatch, rootState }, { key, password }) {
      const seed = Bip39.mnemonicToSeed(key);
      const hdKey = HDKey.fromMasterSeed(seed);
      const hdWallet = hdKey.derivePath(hdKeyMnemonic.path);

      /* eslint-disable no-await-in-loop */
      for (let index = 0; index < 5; index++) {
        const wallet = hdWallet.deriveChild(index).getWallet();
        const walletV3 = wallet.toV3(Buffer.from(password), kdfParams);
        const { address } = walletV3;

        if (index !== 0) {
          dispatch('addWallet', walletV3);
        } else {
          dispatch('addWalletAndSelect', walletV3);
        }

        try {
          const balance = await rootState.web3.web3.eth.getBalance(address);

          if (balance === '0') {
            break;
          }
        } catch (e) {
          break;
        }
      }
      /* eslint-enable no-await-in-loop */
    },
    updateBalance({ commit, dispatch, state, rootState }) {
      if (state.address) {
        const address = state.address.getChecksumAddressString();

        return rootState.web3.web3.eth
          .getBalance(address)
          .then(balance => commit('setBalance', balance))
          .catch(e => dispatch('errors/emitError', e, { root: true }));
      }
    },
    updateSettings({ commit, dispatch }, settings) {
      commit('setSettings', settings);
      return storage
        .write('settings', settings)
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    validatePassword({ state }, password) {
      return state.wallet.validatePassword(password);
    },
    login({ commit, dispatch }, email) {
      return userService.login(email);
    },
    logout({ commit, dispatch }) {
      commit('setEmail', null);
      return Promise.all([storage.clear(), userService.logout()])
        .then(() => window.location.reload())
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    loginViaOTP({}, { code, email }) {
      return userService.loginViaOTP(code, email);
    },
    getOtpSettings({ commit, dispatch }) {
      return userService
        .getOtpSettings()
        .then(otpSettings => commit('setOtpSettings', otpSettings))
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    setOtpSettings({ commit, dispatch }, { secret, code }) {
      return userService
        .setOtpSettings(secret, code)
        .then(() => commit('setOtpSettings', { status: 'enabled' }))
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    deleteOtpSettings({ commit, dispatch }, { code }) {
      return userService
        .deleteOtpSettings(code)
        .then(() => {
          commit('setOtpSettings', {});
          dispatch('getOtpSettings');
        })
        .catch(e => dispatch('errors/emitError', e, { root: true }));
    },
    async init({ commit, dispatch }) {
      try {
        let [settings, email] = await Promise.all([
          userService.getSettings(),
          storage.read('email'),
        ]);

        commit(`tokens/${SAVE_TOKENS}`, settings.tokens || {}, { root: true });
        commit('setEmail', email);

        if (settings) {
          commit('setSettings', settings);
        }

        if (!email) {
          storage.disableRemote();
          return null;
        }

        // Fetch and save HD wallet
        let hdKey = await userService.getHDKey();
        if (hdKey) {
          commit('setHdKey', hdKey);
        }

        // Fetch and save regular accounts
        let accounts = await userService.getV3Accounts();
        if (accounts && accounts.length) {
          accounts.forEach(wallet => commit('addWallet', wallet));
          await dispatch('selectWallet', accounts[0].address);
        }
      } catch (e) {
        return dispatch('errors/emitError', e, { root: true });
      }
    },
  },
};
