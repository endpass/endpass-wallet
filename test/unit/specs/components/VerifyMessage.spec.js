import Vuex from 'vuex';
import { shallow, createLocalVue } from '@vue/test-utils';
import Notifications from 'vue-notification';
import VerifyMessage from '@/components/VerifyMessage';
import ethereumWalletMock from 'fixtures/wallet';
import { address } from 'fixtures/accounts';
import web3 from '@/class/singleton/web3';

const localVue = createLocalVue();

localVue.use(Notifications);
localVue.use(Vuex);

describe('VerifyMessage', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        accounts: {
          namespaced: true,
          getters: {
            wallet: () => ethereumWalletMock,
          },
        },
      },
      state: {
        web3: {},
      },
    });

    wrapper = shallow(VerifyMessage, {
      localVue,
      store,
    });
  });

  describe('render', () => {
    it('should render the initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should enable "Verify message" button', () => {
      wrapper.setData({
        signedMessageString: '{}',
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should disable "Verify message" button', () => {
      wrapper.setData({
        signedMessageString: 'string',
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should show the received address', () => {
      wrapper.setData({ address });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('verifyMessage', () => {
      beforeEach(() => {
        jest.spyOn(wrapper.vm, '$notify');
      });

      it('should verify message', async () => {
        expect.assertions(2);

        const { vm } = wrapper;

        web3.eth.accounts.recover = jest.fn().mockResolvedValueOnce(address);
        wrapper.setData({
          signedMessageString: '{}',
        });

        await vm.verifyMessage();

        expect(vm.address).toEqual(ethereumWalletMock.getAddress());
        expect(vm.$notify).not.toHaveBeenCalled();
      });

      it('should not verify message', async () => {
        expect.assertions(3);

        const { vm } = wrapper;

        web3.eth.accounts.recover = jest.fn(() => {
          throw new Error();
        });

        global.console.error = jest.fn();

        await vm.verifyMessage();

        expect(vm.address).toBeNull();
        expect(vm.$notify).toHaveBeenCalledTimes(1);
        expect(vm.$notify).toHaveBeenCalledWith({
          title: 'Error while verifying the message',
          text:
            'An error occurred while verifying the message. Please try again.',
          type: 'is-danger',
        });
      });
    });
  });
});
