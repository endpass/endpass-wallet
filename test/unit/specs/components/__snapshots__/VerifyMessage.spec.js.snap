// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyMessage render should disable "Verify message" button 1`] = `
<v-form>
  <v-textarea
    label="Signature"
    v-model="signedMessageString"
  />
   
  <v-button
    classname="is-primary is-medium"
    disabled="true"
  >
    Verify message
  </v-button>
   
  <!---->
</v-form>
`;

exports[`VerifyMessage render should enable "Verify message" button 1`] = `
<v-form>
  <v-textarea
    label="Signature"
    v-model="signedMessageString"
  />
   
  <v-button
    classname="is-primary is-medium"
  >
    Verify message
  </v-button>
   
  <!---->
</v-form>
`;

exports[`VerifyMessage render should render the initial state of the component 1`] = `
<v-form>
  <v-textarea
    label="Signature"
    v-model="signedMessageString"
  />
   
  <v-button
    classname="is-primary is-medium"
    disabled="true"
  >
    Verify message
  </v-button>
   
  <!---->
</v-form>
`;

exports[`VerifyMessage render should show the received address 1`] = `
<v-form>
  <v-textarea
    label="Signature"
    v-model="signedMessageString"
  />
   
  <v-button
    classname="is-primary is-medium"
    disabled="true"
  >
    Verify message
  </v-button>
   
  <div
    class="field"
  >
    <label
      class="label"
    >
      Address
    </label>
     
    <p>
      address
    </p>
  </div>
</v-form>
`;
